java-generator/core/src/main/java/io/fabric8/java/generator/nodes/JObject.java

```diff
@@ -94,10 +95,41 @@ public JObject(
       for (Map.Entry<String, JSONSchemaProps> field : fields.entrySet()) {
         String nextPrefix = (config.getPrefixStrategy() == Config.Prefix.ALWAYS) ? classPrefix : "";
         String nextSuffix = (config.getSuffixStrategy() == Config.Suffix.ALWAYS) ? classSuffix : "";
+        String fieldKey = field.getKey();
+        // let's handle duplicates
+        final Map<String, Map<String, JSONSchemaProps>> duplicates = new HashMap<>();
+        // counting the current field duplicates
+        final List<Map.Entry<String, JSONSchemaProps>> fieldDuplicates = fields.entrySet().stream()
+            .filter(f -> sanitizeString(field.getKey()).equals(sanitizeString(f.getKey())))
+            .collect(Collectors.toList());
+        final Long duplicatesCount = fieldDuplicates.stream().count();
+        final Boolean duplicatesExist = duplicatesCount > 1;
+        if (duplicatesExist) {
+          // we want to throw an exception on some duplicates missing requirements. At the moment, the only one we
+          // enforce is for a (duplicatesCount - 1) number of duplicates to be marked as deprecated
+          final Long deprecatedDuplicates = fieldDuplicates.stream()
+              .filter(d -> d.getValue().getDescription().startsWith(DEPRECATED_FIELD_MARKER)).count();
```

Comment:
I would keep this a little more relaxed like:

```
getDescription().toLowercase().contains("deprecated")
```
wdyt?

Reply:
I am not sure, to be honest. Any description containing such string would mark the field as deprecated, even when it is not. I'd be more strict here and keep it as is. That being said, it would need a correspondent approach in crd-generator.
WDYT?

Reply:
Anything we do here is already well beyond the compatibility with the crd-gen ...
I would not be worried as we are handling edge cases.
I think we can agree on a compromise:
- keep `startsWith` but make sure we trim first the string
- lower the requirement using `toLowerCase` and match on `deprecated`

I do believe this is a reasonable compromise :-)

Reply:
+1, thanks, fixing!